vortex-file/src/generic.rs

```diff
@@ -42,10 +43,18 @@ impl VortexOpenOptions<GenericVortexFile> {
     pub async fn open<R: VortexReadAt + Send>(self, read: R) -> VortexResult<VortexFile> {
         let footer = self.read_footer(&read).await?;
 
+        let segment_cache = Arc::new(SegmentCacheMetrics::new(
+            InitialReadSegmentCache {
+                initial: self.initial_read_segments,
```

Comment:
do we never wanna expire these, no matter how high the initial read might be set?

Reply:
Not sure, I toyed with these being in the actual segment cache. In the end, figured if the user controls the size of the initial read, it's easy to understand that memory is held while the VortexFile is open?